### DISCUSSION TITLE
External Storage Support

### DISCUSSION THREAD

**Discussion by shyim** [bolt/core#1681] [Ideas / Feature requests]
* Usage of Flysystem for all FS operations. 
* Migrate current config to an local adapter
* Allow usage of other adapters like S3, GCP with config

**Comment by bobdenotter** [comment]
Just a thought: If we do this, should we just go ahead, and integrate Flysystem v2 (which is in beta)? 

Also, I've personally not used external storage often. Would you like to help out with actually implementing it?

**Comment by LewisHobden** [comment]
Hi! I would be interested in helping to implement this. I've had plenty of experience with Flysystem and this feels like a valuable feature for the project.

Would I need to look for a sponsor for this kind of feature request before it can start?

**Comment by bobdenotter** [comment]
Hi, @LewisHobden 👋 

Awesome! And welcome! You won't need an official sponser. As you'll find out, we're a fairly easy-going bunch. That said, it's obviously good to discuss the approach to see if it's a good way to do it, before digging in. :-)

How'd you like to approach this? Perhaps you could open a separate `[RFC] Flysystem` issue, where you briefly describe how/what it should do, and what you intend to implement. From there, we can help you get started and find your way around the code. Would that work?

**Comment by LewisHobden** [comment]
That sounds fine by me, I've created it at #1929, hopefully that works!
